profiles-paper-gui/src/main/java/com/awooga/profiles/dao/impl/PlayerProfilesDAOImpl.java

```diff
@@ -118,14 +119,15 @@ public boolean deleteProfile(Player player, UUID profileUuid) {
 	@SneakyThrows
 	@Override
 	public List<@NotNull ProfileEntity> getProfileEntitiesByGenuineUUID(UUID genuineUuid) {
-		PreparedStatement stmt = conn.prepareStatement("SELECT * FROM profiles WHERE playerUuid=? AND deleted=false");
+		PreparedStatement stmt = conn.prepareStatement("SELECT * FROM profiles WHERE playerUuid=? AND deleted=false"
+			, ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_READ_ONLY);
 		stmt.setString(1, genuineUuid.toString());
 		ResultSet resultSet = stmt.executeQuery();
 		List<ProfileEntity> result = new ArrayList();
 
 		if(!config.getBoolean("options.disableMojangProfile", true)) {
 			ProfileEntity ent = ProfileEntity.builder()
-				.id(resultSet.getLong("id"))
+				.id(-1L)
```

Comment:
Is this because of an empty result set NPE or similar?

Reply:
Even if the resultSet was empty (player has no profiles) it would still try to grab an id from the resultSet even when there are no rows to grab from. At best, it would simply throw an error. At worst, it could grab the id from another profile and cause issues that way. I replaced it with -1, because this ID doesn't seem important after the ProfileEntity builder.